File: DSA/DSA Using Go/Binary_Search_Recursive.go
package main

import (
	"fmt"
)

func BinarySearch(slice []int, element, left, right int) int {
	var m int
	if left > right {
		return -1
	} else {
		m = (left + right) / 2
		if slice[m] == element {
			return m
		} else if element < slice[m] {
			return BinarySearch(slice, element, left, m-1)
		} else if element > slice[m] {
			return BinarySearch(slice, element, m+1, right)
		}
	}
	return -1
}

func main() {
	var slice = []int{12, 45, 65, 76, 83}
	op := BinarySearch(slice, 100, 0, len(slice)-1)
	fmt.Println(op)
}

File: DSA/DSA Using Python/Binary_Search_Recursive.py
import time
def BinarySearch(list, element, left, right):
    if left > right:
        return -1
    else:
        mid = (left+right)//2
        if element == list[mid]:
            return mid
        elif element < list[mid]:
            return BinarySearch(list, element, left, mid-1)
        elif element > list[mid]:
            return BinarySearch(list, element, mid+1, right)

li = [2,3,7,12,17,23,45,70]
t = time.time()
op = BinarySearch(li, 45, 0, len(li)-1 )
print(op)
print(time.time() - t)
        